client/core/bond.go

```diff
@@ -170,7 +338,76 @@ func (c *Core) refundExpiredBonds(ctx context.Context) {
 				}
 			}
 		}
+		expiredStrength := sumBondStrengths(dc.acct.expiredBonds)
 		dc.acct.authMtx.Unlock()
+
+		if authed && mustPost > 0 && targetTier > 0 && bondExpiry > 0 {
+			c.log.Infof("Gotta post %d bond increments now. Target tier %d, current tier %d (%d weak, %d pending)",
+				mustPost, targetTier, tier, weakStrength, pendingStrength)
+			if bondKey == nil || dc.status() != comms.Connected {
+				c.log.Warnf("Unable to post the required bond while disconnected or logged out.")
+				continue
+			}
+
+			bondAsset := bondAssets[bondAssetID]
+			if bondAsset == nil {
+				c.log.Warnf("Bond asset %d not supported by DEX %v", bondAssetID, dc.acct.host)
+				continue
+			}
+
+			wallet, err := c.connectedWallet(bondAssetID)
+			if err != nil {
+				c.log.Errorf("%v wallet not available to post bond: %v", unbip(bondAssetID), err)
+				continue
+			}
+			if _, ok := wallet.Wallet.(asset.Bonder); !ok { // will fail in MakeBondTx, but assert here anyway
+				c.log.Errorf("Wallet %v is not an asset.Bonder", unbip(bondAssetID))
+				continue
+			}
+			_, err = wallet.refreshUnlock()
+			if err != nil {
+				c.log.Errorf("failed to unlock bond asset wallet %v: %v", unbip(bondAssetID), err)
+				continue
+			}
+
+			// For the max bonded limit, we'll normalize all bonds to the
+			// currently selected bond asset.
```

Comment:
The idea of max bonded is to allow overcoming penalization, otherwise your account can get suspended (go sub zero tier) and it won't post any additional bond to get into a trading tier.

Reply:
> Another thought this brings to mind is, if bond amount is changed, ideally, the user is not punished. If a full bond was bought yesterday, and the bond price goes up today, that user should not need to buy more bonds for the same tier, imo, which I believe happens here.

We can't address this easily.  Server's gonna apply the new bond amount to existing bonds when user authorizes.